src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs

```diff
@@ -1187,6 +1187,12 @@ public ErrorDiagnostic UnknownModuleReferenceScheme(string badScheme, ImmutableA
                 TextSpan,
                 "BCP200",
                 $"The specified OCI artifact reference \"{badRef}\" is not valid. The registry \"{badRegistry}\" exceeds the maximum length of {maxLength} characters.");
+
+            public Diagnostic InvalidBicepConfigurationFile(string configurationPath, string errorMessage) => new(
+                TextSpan,
+                DiagnosticLevel.Warning,
+                "BCP201",
+                $"The bicep configuration file \"{configurationPath}\" is not valid: {errorMessage}");
```

Comment:
I think it might be worth enumerating possible cases, or separating them out in the exception handling in case the error message coming back from the library is not clear. E.g. "The Bicep configuration file \"{configurationPath}\" could not be loaded..." & "Failed to parse Bicep configuration file \"{configurationPath}\"..."